build-support/bin/release.sh

```diff
@@ -614,59 +662,34 @@ function publish_packages() {
   end_travis_section
 }
 
-function usage() {
-  echo "With no options all packages are built, smoke tested and published to"
-  echo "PyPi.  Credentials are needed for this as described in the"
-  echo "release docs: http://pantsbuild.org/release.html"
-  echo
-  echo "Usage: $0 [-d] [-c] (-h|-n|-f|-t|-l|-o|-e|-p)"
-  echo " -d  Enables debug mode (verbose output, script pauses after venv creation)"
-  echo " -h  Prints out this help message."
-  echo " -n  Performs a release dry run."
-  echo "       All package distributions will be built, installed locally in"
-  echo "       an ephemeral virtualenv and exercised to validate basic"
-  echo "       functioning."
-  echo " -f  Build the fs_util binary."
-  echo " -t  Tests a live release."
-  echo "       Ensures the latest packages have been propagated to PyPi"
-  echo "       and can be installed in an ephemeral virtualenv."
-  echo " -l  Lists all pantsbuild packages that this script releases."
-  echo " -o  Lists all pantsbuild package owners."
-  echo " -e  Check that wheels are prebuilt for this release."
-  echo " -p  Build a pex from prebuilt wheels for this release."
-  echo " -q  Build a pex which only works on the host platform, using the code as exists on disk."
-  echo
-  echo "All options (except for '-d') are mutually exclusive."
-
-  if (( $# > 0 )); then
-    die "$@"
-  else
-    exit 0
-  fi
-}
-
-while getopts "hdnftcloepqw" opt; do
```

Comment:
This is all fine, but you could also just parse options in two passes:
```bash
_OPTS="hdnftcloepqw3"

while getopts "${_OPTS}"  opt; do
  case ${opt} in
    3) export PY=bob ;;
  esac
done

OPTIND=0
while getopts "${_OPTS}" opt; do
  ... as exists
done
```

If you like this better - I think you can revert this file from the PR since the change to add a -3 option will then be self-contained.

Reply:
I slightly prefer the PR's version for consistency with `ci.sh`, but I'll go with your proposal as it fits the file better - for example, we already have `run_packages_script` so it is confusing to add all these new `run_list`, `run_list_with_ownership`, etc checks.

Once again you taught me something new about Bash! Thanks for the suggestion!